packages/sitecore-jss/src/graphql-request-client.ts

```diff
@@ -85,41 +90,74 @@ export type GraphQLRequestClientFactoryConfig = {
   apiKey?: string;
 };
 
+/**
+ * Checks if the provided error is a ClientError.
+ * @param {RetryError} error - The error to be checked.
+ * @returns {boolean} - True if the error is a ClientError, false otherwise.
+ */
+function isClientError(error: ClientError | NodeJS.ErrnoException): error is ClientError {
```

Comment:
I discussed this with Adam but then we would have to include it in our `RetryStrategy` type and we cant even make it optional since we need them in the `GraphqlRequestClient` class to know the `status` and `response` for debug logging. Also not making them optional would mean a big breaking change which we want to avoid.

Reply:
I think @illiakovalenko is proposing these typeguard functions should be protected functions on our `DefaultRetryStrategy` (so they could be re-used by any class extending it). However, I'm not sure extending that class is all that useful other than to get those. We could add an abstract base class which includes these but that seems overkill. I'm also not sure it's worth the noise of exporting these functions publicly standalone. 

Do you think it's too cumbersome for folks to check types on their own?